File: examples/class-kits/gospider/infinite_scroll.go
package classkit

import scrapyapi "gospider/scrapy"

func NewInfiniteScrollSpider() *scrapyapi.Spider {
	return scrapyapi.NewSpider("infinite-scroll", func(response *scrapyapi.Response) ([]any, error) {
		return []any{
			scrapyapi.NewItem().
				Set("kind", "infinite_scroll").
				Set("title", response.CSS("title").Get()).
				Set("url", response.URL),
		}, nil
	}).
		AddStartURL("https://example.com/discover").
		WithStartMeta("runner", "browser")
}

File: examples/class-kits/rustspider/infinite_scroll.rs
use std::sync::Arc;

use rustspider::scrapy::project as projectruntime;
use rustspider::scrapy::{Item, Output, Response, Spider};

pub fn make_infinite_scroll_spider() -> Spider {
    Spider::new(
        "infinite-scroll",
        Arc::new(|response: &Response| {
            vec![Output::Item(
                Item::new()
                    .set("kind", "infinite_scroll")
                    .set("title", response.css("title").get().unwrap_or_default())
                    .set("url", response.url.clone()),
            )]
        }),
    )
    .add_start_url("https://example.com/discover")
    .with_start_meta("runner", "browser")
}

pub fn register() {
    projectruntime::register_spider("infinite-scroll", make_infinite_scroll_spider);
}

